api/v1alpha1/serviceintentions_types.go

```diff
@@ -187,6 +272,32 @@ func (in *ServiceIntentions) Validate() error {
 	return nil
 }
 
+func (in IntentionPermissions) validate(path *field.Path) field.ErrorList {
+	var errs field.ErrorList
+	for i, permission := range in {
+		if err := permission.Action.validate(path.Child("permissions").Index(i)); err != nil {
+			errs = append(errs, err)
+		}
+		if permission.HTTP != nil {
+			if err := permission.HTTP.validate(path.Child("permissions").Index(i)); err != nil {
+				errs = append(errs, err...)
+			}
+		}
+	}
+	return errs
+}
+
+func (in *IntentionHTTPPermission) validate(path *field.Path) field.ErrorList {
+	var errs field.ErrorList
+	if invalidPathPrefix(in.PathPrefix) {
+		errs = append(errs, field.Invalid(path.Child("pathPrefix"), in.PathPrefix, `must begin with a '/'`))
+	}
+	if invalidPathPrefix(in.PathExact) {
+		errs = append(errs, field.Invalid(path.Child("pathExact"), in.PathExact, `must begin with a '/'`))
+	}
+	return errs
```

Comment:
do you think we need to validate that a valid HTTP method was provided? This is not a blocker by any means, could be added at a later time once we have more docs and stuff from consul.

Reply:
I was wondering the same thing. I wasn't sure if casing was a problem or if wild cards were supported which I why I punted on that for now. If the exhaustive list of methods is GET, PUT, POST, PATCH & DELETE (do you know if we support CONNECT, OPTIONS, TRACE and HEAD)??

Reply:
PS: i had to look up MDN to get the exhaustive list of http methods 😛

Reply:
Ah, fair enough! Definitely makes sense to punt! Haha, I don't even know what is MDN 🤯

Reply:
Thanks for the review @ishustava 💥